### PR TITLE
Hotfix for cell rotations and translations

### PR DESCRIPTION
**Pull request by GiudGiud**
Cell rotations were only partially implemented
- finding cells was correct
- finding the distance to the next surface did not account for rotations or translations, and could not since it was looking at coord from most local to most global (unlike in the develop branch)
- finding a maxXYZ of a rotated cell is still incorrect. I added a warning log. The log is set to DEBUG in the Z-direction to avoid log clutter in 3D simulations. The problem is that the maxX of a cell with an XPlane boundary is easily found, but if that XPlane is rotated by pi/4, then the intersection with other planes has to be considered.

Other changes
- bug fix for computing the rotation matrix from rotation angles
- checks on small numbers in CMFD have been adapted to the case of the BEAVRS core. Beyond the RPV  fluxes get very small and the reaction tallies are small. This triggers a check that sets them to 1E-10. But the CMFD matrix elements are proportional to cross sections and are not actually small.
If CMFD diagonal matrix elements go to zero (unphysical) due to those checks failing, the linear solver fails. 
- small fixes on syntax of if and for loops
- small fixes on output of getPhi/Psi/Theta routines in degrees

### CODE REVIEW

**Comment by GiudGiud** [comment]
Adding unit tests for geometry.findCell and geometry.findNextCell to avoid the issue re-appearing

File: tests/test_find_cell/test_find_cell.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        core = openmoc.Lattice(name='2x2 core')
+        core.setWidth(width_x=2.0, width_y=2.0)
+        core.setUniverses([[[assembly, assembly_rt], [assembly_t, assembly_r]]])
+        root_cell.setFill(core)
+
+        self.geometry = openmoc.Geometry()
+        self.geometry.setRootUniverse(root_universe)
+
+        # To debug the test
+        if False:
```
Comment (wuwenbin2006) [inline tests/test_find_cell/test_find_cell.py]: To be deleted.

File: tests/test_find_cell/test_find_cell.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        assembly_c.setFill(assembly)
+        assembly_c.setTranslation((0.05,0,0))
+        assembly_t = openmoc.Universe(name='translated lattice')
+        assembly_t.addCell(assembly_c)
+
+        # Rotate a lattice to test rotation of a fill cell
+        assembly_c = openmoc.Cell(name='translated cell containing a lattice')
+        assembly_c.setFill(assembly)
+        assembly_c.setRotation((0,0,-90), 'degrees')
+        #assembly_c.setTranslation((-1,0,0)) # doesn't seem to be supported
```
Comment (wuwenbin2006) [inline tests/test_find_cell/test_find_cell.py]: I think I located the bug. They are lines 542-544 in Universe.cpp.
They should be 
          double x = next_coords->getX();
          double y = next_coords->getY();
          double z = next_coords->getZ();

File: src/Cell.cpp [hunk 3]
```diff
@@ -904,7 +934,7 @@ void Cell::setRotation(double* rotation, int num_axes, std::string units) {
   _rotation_matrix[0] = cos(theta) * cos(phi);
   _rotation_matrix[1] = cos(theta) * sin(phi);
   _rotation_matrix[2] = -sin(theta);
-  _rotation_matrix[3] = sin(psi) * sin(theta) * cos(psi) -
+  _rotation_matrix[3] = sin(psi) * sin(theta) * cos(phi) -
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply src/Cell.cpp]: I checked the other elements

File: src/Cell.cpp [hunk 4]
```diff
@@ -382,6 +382,10 @@ int Cell::getNumSectors() {
  */
 double Cell::getMinX() {
 
+  if (_rotated || _translated)
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply src/Cell.cpp]: The convention in the code so far is that the cell **fill** is rotated. When querying the minX of a cell, we are only considering the region. So the result is correct with regards to the convention, but not intuitive.

File: tests/test_find_cell/test_find_cell.py (hunk 2 above)
Comment (GiudGiud) [reply tests/test_find_cell/test_find_cell.py]: Good catch ! I saw these lines 1,000 times while doing this PR yet it didn't click

File: tests/test_find_cell/test_find_cell.py (hunk 1 above)
Comment (GiudGiud) [reply tests/test_find_cell/test_find_cell.py]: done

**Pull request merged**